relay-server/src/services/project_upstream.rs

```diff
@@ -394,7 +400,17 @@ impl UpstreamProjectSourceService {
                             counter(RelayCounters::ProjectUpstreamCompleted) += 1,
                             result = result,
                         );
-                        channel.send(state.sanitize());
+
+                        if invalid {
+                            // Treat invalid as pending, try again:
+                            // NOTE: We might want to implement a backoff here, because the
+                            // chance that an invalid config will turn into a valid config
+                            // within `query_interval` is low.
```

Comment:
This is the most high-risk change in this PR: By treating pending and invalid the same, we will keep requesting invalid configs. Might be worth changing this behavior back to use `project.next_fetch_attempt` for unparsable configs instead.

Reply:
Reverted, we will keep special-casing invalid project configs with the outer backoff loop for now (see `Project::backoff`.